src/lib.rs

```diff
@@ -305,10 +307,26 @@ where
                 text_for_header.push_str(text);
                 text_for_header.push('`');
             }
+
+            // (re)-escape `|` when it appears as part of inline code in the
+            // body of a table. NOTES:
+            // - This is always *safe*, but may not be *necessary*: more recent
+            //   versions of `pulldown-cmark` do not require this because they
+            //   correctly handle unescaped pipes in this position.
```

Comment:
I'd like to learn more. All I know is that v0.12 of `pulldown-cmark` unescapes the content, so indeed we have to escape it again and we know it as we control the version we use.

Reply:
Ah, looking at this again, I simply mistook the results I was seeing—running `pulldown-cmark` *directly* is fine, because it’s not emitting Markdown again, but using it to round trip the way this library does needs re-escaping. I’ll remove the comment!